api/src/middleware/extract-token.ts

```diff
@@ -14,17 +14,19 @@ import { useEnv } from '@directus/env';
 const extractToken: RequestHandler = (req, _res, next) => {
 	const env = useEnv();
 
-	let token: string | null = null;
+	req.token = null;
 
-	if (req.query && req.query['access_token']) {
-		token = req.query['access_token'] as string;
+	if (req.query['access_token'] && typeof req.query['access_token'] === 'string') {
+		req.token = req.query['access_token'];
+		req.tokenSource = 'query';
 	}
 
-	if (req.headers && req.headers.authorization) {
+	if (req.headers.authorization) {
 		const parts = req.headers.authorization.split(' ');
+		const [scheme, token] = parts;
 
-		if (parts.length === 2 && parts[0]!.toLowerCase() === 'bearer') {
-			if (token !== null) {
+		if (parts.length === 2 && scheme?.toLowerCase() === 'bearer' && token) {
```

Comment:
Could be further simplified to 

```ts
const [schema, token] = req.headers.authorization.split(' ');

if (scheme?.toLowerCase() === 'bearer' && token)
```

Reply:
I think we should keep it strict here and only match if there are exactly two parts.

Reply:
But that's already happening with the `&& token`. `token` is `undefined` if split returns only one part.

Reply:
Correct, but it would allow > 2 parts, `bearer my-token some-invalid-thing`

Reply:
Very true!

Reply:
We could simplify it as
```ts
const [schema, token, ...rest] = req.headers.authorization.split(' ');

if (rest.length === 0 && scheme?.toLowerCase() === 'bearer' && token)
```
But not sure if it's worth it 🤔